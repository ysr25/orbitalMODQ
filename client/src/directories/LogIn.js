import React, { Component } from "react";
import { Link } from "react-router-dom";
import Button from "react-bootstrap/Button";
import settings from "../config/settings";
import Form from "react-bootstrap/Form";
import Col from "react-bootstrap/Col";

export default class LogIn extends Component {
  constructor() {
    super();

    this.state = {
      username: "",
      password: "",
      isButtonDisabled: false,
      status: null,
    };
  }

  onChangeUsername = (e) => {
    this.setState({
      username: e.target.value,
      isButtonDisabled: false,
    });
  };

  onChangePassword = (e) => {
    this.setState({
      password: e.target.value,
      isButtonDisabled: false,
    });
  };

  onSubmit = (e) => {
    e.preventDefault();

    const userLogIn = {
      username: this.state.username,
      password: this.state.password,
    };

    this.setState({
      isButtonDisabled: true,
      status: "Logging In...",
    });

    this.props
      .api("post", "/users/login", userLogIn)
      .then((res) => {
        this.setState({ status: res.data.message });
        this.props.history.push("/");
      })
      .catch((err) => {
        this.setState({
          status: err.response.data.message,
          isButtonDisabled: true,
        });
      });
  };

  render() {
    return (
      <>
        <h3>Log in</h3>
        <p>
          Don't have an account? <Link to="/users/signup">Register</Link>
        </p>
        <Form onSubmit={this.onSubmit}>
          <Form.Row>
            <Form.Label htmlFor="username" column sm={2}>
              Username
            </Form.Label>
            <Col>
              <Form.Control
                id="username"
                type="text"
                value={this.state.username}
                onChange={this.onChangeUsername}
                required
              />
            </Col>
          </Form.Row>
          <br />

          <Form.Row>
            <Form.Label htmlFor="password" column sm={2}>
              Password
            </Form.Label>
            <Col>
              <Form.Control
                id="password"
                type="password"
                value={this.state.password}
                onChange={this.onChangePassword}
                required
              />
            </Col>
          </Form.Row>
          <br />

          <Button type="submit" disabled={this.state.isButtonDisabled}>
            Log In
          </Button>
          <br />
          {this.state.status}
          <br />
          <a
            href={settings.serverUrl + "api/users/login/google"}
            className="btn btn-primary"
          >
            Log in with Google
          </a>
        </Form>
      </>
    );
  }
}
