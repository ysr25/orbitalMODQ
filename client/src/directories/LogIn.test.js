import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import LogIn from "./LogIn";

it("renders login page", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <LogIn />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
