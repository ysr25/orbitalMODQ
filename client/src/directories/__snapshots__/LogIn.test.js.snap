// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders login page 1`] = `
Array [
  <h3>
    Log in
  </h3>,
  <p>
    Don't have an account? 
    <a
      href="/users/signup"
      onClick={[Function]}
    >
      Register
    </a>
  </p>,
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-row"
    >
      <label
        className="form-label col-form-label col-sm-2"
        htmlFor="username"
      >
        Username
      </label>
      <div
        className="col"
      >
        <input
          className="form-control"
          id="username"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <br />
    <div
      className="form-row"
    >
      <label
        className="form-label col-form-label col-sm-2"
        htmlFor="password"
      >
        Password
      </label>
      <div
        className="col"
      >
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
    </div>
    <br />
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Log In
    </button>
    <br />
    <br />
    <a
      className="btn btn-primary"
      href="http://localhost:3001/api/users/login/google"
    >
      Log in with Google
    </a>
  </form>,
]
`;
